python/ordinal-classifier.ipynb

```diff
@@ -0,0 +1,466 @@
+    "ordinal_clf.fit(X_train_transformed, y_train)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Make predictions\n",
+    "y_pred = ordinal_clf.predict(X_test_transformed)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Evaluate model performance - MAE and RMSE\n",
+    "mae = mean_absolute_error(y_test, y_pred, multioutput='raw_values')\n",
+    "mse = mean_squared_error(y_test, y_pred, multioutput='raw_values')\n",
+    "rmse = np.sqrt(mse)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Mean Absolute Error:  [0.31698113 0.34716981 0.52075472 0.37358491 0.38490566 0.46037736\n",
+      " 1.0490566 ]\n",
+      "Root Mean Squared Error:  [0.56301077 0.59558121 0.74225561 0.62344145 0.6672953  0.68405009\n",
+      " 1.50219965]\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(\"Mean Absolute Error: \", mae)\n",
+    "print(\"Root Mean Squared Error: \", rmse)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## What are MAE and RMSE, and what do they tell me?\n",
+    "The **Mean Absolute Error (MAE)** and **Root Mean Squared Error (RMSE)** are metrics used to evaluate the performance of a regression model. In this case, we have an ordinal classification model, which is a form of regression, so these metrics are appropriate.\n",
+    "\n",
+    "**Mean Absolute Error (MAE)** is the average of the absolute differences between the predicted and actual values. It represents the average error magnitude without considering the direction. A *lower* MAE value indicates better performance of the model.\n",
+    "\n",
+    "**Root Mean Squared Error (RMSE)** is the square root of the average of the squared differences between the predicted and actual values. RMSE gives more weight to larger errors, so it is more sensitive to outliers than MAE. A *lower* RMSE value indicates better performance of the model.\n",
+    "\n",
+    "### Interpretation of MAE and RMSE results\n",
+    "The arrays correspond to the seven rating variables in the same order as they appear in the `target_columns` list. The first six rating variables are indicators of decision making, and the last variable is the final decision-making score.\n",
+    "\n",
+    "The results says that the model has a lower error for the first six rating variables compared to the final decision-making score. Interesting. So, the model performs better at predicting the six indicators of decision making than predicting the overall final score 😑 The errors for the first six rating variables are in a similar range, which means the model's performance is pretty consistent across those variables.\n",
```

Comment:
As the final score is a composite of all the other scores, finding the final score from the features would be kind of pointless